cli/migrate/api/console.go

```diff
@@ -0,0 +1,265 @@
+package api
+
+// NOTE: Should this be covered by tests?
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/hasura/graphql-engine/cli/migrate"
+)
+
+const (
+	up   = "up"
+	down = "down"
+)
+
+type UpdateMigrationRequest struct {
+	FileName    string `json:"filename"`
+	FileContent string `json:"content"`
+	// NOTE: we could also keep a log of the changes? would that be necessary?
+}
+
+type MigrationData struct {
+	FileContent      string `json:"content"`
+	FileType         string `json:"type"`
+	MigrationVersion int    `json:"version"`
+}
+
+type MigrationDataResponse struct {
+	Version           string                  `json:"version"`
+	UpMigrationData   MigrationData           `json:"up"`
+	DownMigrationData MigrationData           `json:"down"`
+	MigrationStatus   migrate.MigrationStatus `json:"status"`
+}
+
+func convertToUInt64(param string) uint64 {
+	num, err := strconv.ParseUint(param, 10, 64)
+	if err != nil {
+		return 1
+	}
+
+	return num
+}
+
+// FIXME: May need to change this as well
+func getFileType(ext string) string {
+	if ext == ".yml" || ext == ".yaml" {
+		return "yaml"
+	}
+	if ext == ".sql" {
+		return "sql"
+	}
+
+	// TODO: this should also be an error probably
+	return ""
+}
+
+func getMigrationDirName(version string, name string) string {
+	if name == "" {
+		return version
+	}
+
+	return version + "_" + name
+}
+
+func isFilePresent(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+
+	return !info.IsDir()
+}
+
+func getMigrationFileContents(path string) (string, error) {
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+func getValidFilePath(path string, migrationType string, extensions []string) (string, string) {
+	for _, ext := range extensions {
+		fullPath := path + migrationType + ext
+		exists := isFilePresent(fullPath)
+		if exists {
+			return fullPath, ext
+		}
+	}
+	// TODO: this should probably be an error
+	return "", ""
+}
+
+func getMigrationInfo(path string, migrationName string, version int) (MigrationData, MigrationData, error) {
+	// returns the type of file yml, yaml or sql
+	// the respective contents of the file
+	// this function will only look for up.* and down.* for now
+
+	fullPath := path + "/" + migrationName + "/"
+	validExtensions := []string{".yaml", ".sql", ".yml"}
+
+	upSource, upExt := getValidFilePath(fullPath, up, validExtensions)
+	downSource, downExt := getValidFilePath(fullPath, down, validExtensions)
```

Comment:
this would only work if the file names were `up.*` and `down.*`. Need to check if other names are permissible

Reply:
Yes! this is case, we don't have to worry..:)